aspnetcore/security/authentication/identity-configuration.md

```diff
@@ -31,61 +58,58 @@ ASP.NET Core 2.0 added the `RequiredUniqueChars` property. Otherwise, the option
 
-[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,44-46,50-52)]
+### Tokens
 
-`IdentityOptions.SignIn` has the following properties:
+[IdentityOptions.Tokens](/dotnet/api/microsoft.aspnetcore.identity.identityoptions.tokens) specifies the [TokenOptions](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions) with the properties shown in the table.
 
-| Property                | Description                       | Default |
-| ----------------------- | --------------------------------- | ------- |
-| `RequireConfirmedEmail` | Requires a confirmed email to sign in. | false  |
-| `RequireConfirmedPhoneNumber` |  Requires a confirmed phone number to sign in. | false  |
+| Property | Description |
+| -------- | ----------- |
+| [AuthenticatorTokenProvider](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.authenticatortokenprovider) | Gets or sets the `AuthenticatorTokenProvider` used to validate two-factor sign-ins with an authenticator. |
+| [ChangeEmailTokenProvider](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.changeemailtokenprovider) | Gets or sets the `ChangeEmailTokenProvider` used to generate tokens used in email change confirmation emails. |
+| [ChangePhoneNumberTokenProvider](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.changephonenumbertokenprovider) | Gets or sets the `ChangePhoneNumberTokenProvider` used to generate tokens used when changing phone numbers. |
+| [EmailConfirmationTokenProvider](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.emailconfirmationtokenprovider) | Gets or sets the token provider used to generate tokens used in account confirmation emails. |
+| [PasswordResetTokenProvider](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.passwordresettokenprovider) | Gets or sets the [IUserTwoFactorTokenProvider<TUser>](/dotnet/api/microsoft.aspnetcore.identity.iusertwofactortokenprovider-1) used to generate tokens used in password reset emails. |
+| [ProviderMap](/dotnet/api/microsoft.aspnetcore.identity.tokenoptions.providermap) | Used to construct UserTokenProviders with the key used as the providerName. |
 
-## User validation settings
+### User
 
 [!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,48-52)]
 
-`IdentityOptions.User` has the following properties:
+[IdentityOptions.User](/dotnet/api/microsoft.aspnetcore.identity.identityoptions.user) specifies the [UserOptions](/dotnet/api/microsoft.aspnetcore.identity.useroptions) with the properties shown in the table.
 
-| Property                | Description                       | Default |
-| ----------------------- | --------------------------------- | ------- |
-| `RequireUniqueEmail`  | Requires each User to have a unique email. | false  |
-| `AllowedUserNameCharacters`  | Allowed characters in the username. | abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+ |
+| Property | Description |
```

Comment:
Add a "Default" column to the table.

Reply:
It was breaking when the doc was built locally (too wide). I can't see the review site anymore, so I'll change it back and you guys can see if it renders ok.